### PR TITLE
16989 update and create pact tests for covid vaccine

### PR DESCRIPTION
**Pull request by holdenhinkle**
## Description of change
Regarding ticket #16989 - https://app.zenhub.com/workspaces/vsp-5cedc9cce6e3335dc5a49fc4/issues/department-of-veterans-affairs/va.gov-team/16989

These are the Provider tests. The corresponding Consumer tests can be found here - https://github.com/department-of-veterans-affairs/vets-website/pull/15281

- Updates state names for two tests
- Adds two new tests:
  - retrieves previously saved submission data for user
  - submission data for user does not exist

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -1075,4 +1075,4 @@ RUBY VERSION
    ruby 2.6.6p146
 
 BUNDLED WITH
-   1.17.3
+   1.17.3
```
Comment (LindseySaari) [inline Gemfile.lock:1078]: Looks like a typo here

Comment (holdenhinkle) [reply Gemfile.lock:1078]: Somehow an older version of Gemfile.lock got pushed to this branch so I replaced it with the current one in master.

Don't those changes cancel each other out? (I don't know why it looks like that.)

File: spec/service_consumers/provider_states_for/coronavirus_vaccination.rb [hunk 2]
```diff
@@ -1,7 +1,28 @@
 # frozen_string_literal: true
 
 Pact.provider_states_for 'Coronavirus Vaccination' do
-  provider_state 'authenticated user application data' do
+  provider_state 'retrieves previously saved submission data for user' do
+    set_up do
+      user = build_user_and_stub_session
+      submission = CovidVaccine::V0::RegistrationSubmission.for_user(user).last
+      submission || FactoryBot.create(:covid_vaccine_registration_submission)
```
[the remark this answers is not in the document]
Comment (holdenhinkle) [reply spec/service_consumers/provider_states_for/coronavirus_vaccination.rb]: Good catch. Updated it.

File: Gemfile.lock (hunk 1 above)
Comment (LindseySaari) [reply Gemfile.lock:1078]: Is there an extra newline or space possibly?

Comment (holdenhinkle) [reply Gemfile.lock:1078]: No, I checked.

Comment (holdenhinkle) [reply Gemfile.lock]: I just copied/pasted/pushed it again - https://github.com/department-of-veterans-affairs/vets-api/blob/master/Gemfile.lock

**Review by LindseySaari — APPROVED** [review]
👍

**Pull request merged**